### PR TITLE
Feature/fixin tests works

### PR DESCRIPTION
**Pull request by mgloystein**
@bgphipps @brekk @sheeley Please review and merge

### CODE REVIEW

File: src/lib/JSON.lua [hunk 1]
```diff
@@ -0,0 +1,1053 @@
...
+         self:onDecodeError("expected comma or '}'", text, i, etc)
+      end
+
+      i = skip_whitespace(text, i + 1)
+   end
+
+   self:onDecodeError("unclosed '{'", text, start, etc)
+end
+
+local function grok_array(self, text, start, etc)
```
Comment (brekk) [inline src/lib/JSON.lua:565]: :+1:

**Comment by bgphipps** [comment]
LGTM

File: src/lib/JSON.lua [hunk 2]
```diff
@@ -0,0 +1,1053 @@
...
+   end
+
+   if self.assert then
+      self.assert(false, message)
+   else
+      assert(false, message)
+   end
+end
+
+local function grok_number(self, text, start, etc)
```
Comment (brekk) [inline src/lib/JSON.lua:402]: :+1:

**Comment by brekk** [comment]
Generally LGTM; per the standard disclaimer that this ain't my language. :)

**Comment by mgloystein** [comment]
@bgphipps @sheeley

**Comment by bgphipps** [comment]
Merging.

**Pull request merged**